cumulusci/tasks/vlocity/vlocity.py

```diff
@@ -110,46 +110,124 @@ class OmniStudioDeployRemoteSiteSettings(AddRemoteSiteSettings):
+        remote site settings. Function takes a namespace,and injects it
+        into remote site setting url provided.
+
+        A legacy boolean parameter is passed to determine whether to construct
+        the legacy or visualforce url convention.
+
+        An optional org parameter is available to determine the url convention
+        to inject the org type into the remote site setting url
+        to follow the appropriate convention.
+
+        Lastly an optional dns value is given to abstract away the common
+        .my.salesforce.com convention that is common for all instance urls.
+        """
+        if legacy:
+            if org:
+                url = self.org_config.instance_url.replace(
+                    f".{org}{dns}", f"--{namespace}.{org}.vf.force.com"
+                )
+            else:
+                url = self.org_config.instance_url.replace(
+                    f"{dns}", f"--{namespace}.vf.force.com"
+                )
         else:
-            visualforce_url: str = self.org_config.instance_url.replace(
-                ".my.salesforce.com",
-                f"--{namespace}.{self.org_config.instance_name}.visual.force.com",
-            )
-            legacy_visualforce_url: str = self.org_config.instance_url.replace(
-                ".my.salesforce.com",
-                f"--{namespace}.vf.force.com",
-            )
-        lightning_url: str = self.org_config.instance_url.replace(
-            ".my.salesforce.com", ".lightning.force.com"
+            if org:
+                url = self.org_config.instance_url.replace(
+                    f".{org}{dns}",
+                    f"--{namespace}.{org}.{self.org_config.instance_name}.visual.force.com",
+                )
+            else:
+                url = self.org_config.instance_url.replace(
+                    f"{dns}",
+                    f"--{namespace}.{self.org_config.instance_name}.visual.force.com",
+                )
+        return url
+
+    def prepare_remote_site_urls(self, namespace: str, dns: str = ".my.salesforce.com"):
+        """
+        Helper function to create all 3 remote site settings required
+        for Omnistudio Development.
+
+        This includes the following remote site settings:
+
+        - visualforce remote site setting (helper function used)
+        - legacy_visualforce remote site setting (helper function used)
+        - lightning (web component) remote site setting
+
+        Lastly an optional dns value is given to abstract away the common
+        .my.salesforce.com convention that is common for all instance urls.
+        """
+        # developer
+        if f"develop{dns}" in self.org_config.instance_url:
+            visualforce_url = self.create_vf_url(namespace, False, "develop")
+            legacy_visualforce_url = self.create_vf_url(namespace, True, "develop")
+        # patch
+        elif f"patch{dns}" in self.org_config.instance_url:
```

Comment:
I think this could be a loop that tests each prefix.

Or even a regexp just looks for the "overall pattern" and extracts the "patch" or "trailblaze" or "sandbox" part? Maybe it could be future-proofed so that if they invent new ones it will work automatically?